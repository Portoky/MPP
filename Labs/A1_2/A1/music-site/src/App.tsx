import { useState } from "react";
import { BrowserRouter, Routes, Route } from "react-router-dom";
import AddMusic from "./pages/AddMusic";
import Home from "./pages/Home";
import EditMusic from "./pages/EditMusic";
import ViewMusic from "./pages/ViewMusic";
import { createMusic } from "./utils/Utils";
import Diagram from "./pages/Diagram";
const App = () => {
  const [musics, setMusics] = useState([
    createMusic("Rocky Racoon", "The Beatles", 5, 1966),
    createMusic("Nowhere Man", "The Beatles", 4, 1965),
    createMusic("Let It Be", "The Beatles", 4, 1970),
  ]);

  return (
    <div>
      <BrowserRouter>
        <Routes>
          <Route
            path="/"
            element={<Home musics={musics} setMusics={setMusics}></Home>}
          />
          <Route
            path="/edit/:id"
            element={
              <EditMusic musics={musics} setMusics={setMusics}></EditMusic>
            }
          />
          <Route
            path="/view/:id"
            element={<ViewMusic musics={musics}></ViewMusic>}
          />
          <Route
            path="/pages/addmusic"
            element={
              <AddMusic musics={musics} setMusics={setMusics}></AddMusic>
            }
          />
          <Route
            path="/pages/diagram"
            element={<Diagram musics={musics} setMusics={setMusics}></Diagram>}
          />
        </Routes>
      </BrowserRouter>
    </div>
  );
};

export default App;
