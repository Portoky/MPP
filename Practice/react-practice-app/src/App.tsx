import Alert from "./components/Alert";
import Button from "./components/Button";
import { useState } from "react";
//import ListGroup from "./components/ListGroup";

function App() {
  // let items = ["New York", "San Francisco", "Tokyo", "London", "Paris"];

  // //separate event handler
  // const handleSelectItem = (item: string) => {
  //   console.log(item);
  // };
  // return (
  //   <div>
  //     <ListGroup
  //       items={items}
  //       heading="Cities"
  //       onSelectItem={handleSelectItem}
  //     ></ListGroup>{" "}
  //     {/*<Message/> ->self closing*/}
  //   </div>
  // );

  // return (
  //   <div>
  //     <Alert>
  //       Hello<span> Marci!!!</span>
  //     </Alert>
  //     <Button onClick={() => console.log("Clicked!!")} color="secondary">
  //       Gombocska
  //     </Button>
  //   </div>
  // );
  const [alertVisible, setAlertVisible] = useState(false);
  return (
    <div>
      {alertVisible && (
        <Alert onClose={() => setAlertVisible(false)}>My Alert!!</Alert> //!!!!!!!() => ...
      )}
      <Button onClick={() => setAlertVisible(true)} color="secondary">
        My Button
      </Button>
    </div>
  );
}
export default App;
