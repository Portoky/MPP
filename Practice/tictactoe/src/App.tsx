import Game from "./components/Game";

function App() {
  return (
    <>
      <Game></Game>
    </>
  );
}

export default App;
