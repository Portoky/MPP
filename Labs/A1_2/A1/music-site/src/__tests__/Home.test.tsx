import { describe, expect, test } from "vitest";
import { render, screen } from "@testing-library/react";
import App from "../App";
describe("Home", () => {
  test("renders", () => {
    const rendering = render(<App />);
    expect(rendering).toMatchSnapshot();
    expect(screen.getByText("Music forum")).toBeDefined();
  });
});
