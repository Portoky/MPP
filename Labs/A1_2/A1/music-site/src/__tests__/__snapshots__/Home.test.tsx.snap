// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home > renders 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="upperPart"
          style="margin-left: 10px; margin-bottom: 15px;"
        >
          <h1>
            Music forum
          </h1>
          <h5>
            Add, Modify, Remove melodies!
          </h5>
          <div>
            <button
              class="btn btn-outline-primary"
              type="button"
            >
              Add music!
            </button>
             
            <br />
            <button
              class="btn btn-outline-info"
              type="button"
            >
              Check Stats!
            </button>
             
            <br />
            <input
              id="filter"
              name="filter"
              placeholder="Filter by title!"
              type="text"
              value=""
            />
          </div>
        </div>
        <ul
          class="list-group"
          data-testid="list"
        >
          <li
            class="list-group-item list-group-item-dark"
          >
            The Beatles: Rocky Racoon - 1966
            <br />
            <button
              class="btn btn-warning"
              type="button"
            >
              Edit music
            </button>
            <button
              class="btn btn-info"
              type="button"
            >
              View music
            </button>
            <button
              class="btn btn-danger"
              type="button"
            >
              Delete
            </button>
          </li>
          <li
            class="list-group-item list-group-item-dark"
          >
            The Beatles: Nowhere Man - 1965
            <br />
            <button
              class="btn btn-warning"
              type="button"
            >
              Edit music
            </button>
            <button
              class="btn btn-info"
              type="button"
            >
              View music
            </button>
            <button
              class="btn btn-danger"
              type="button"
            >
              Delete
            </button>
          </li>
          <li
            class="list-group-item list-group-item-dark"
          >
            The Beatles: Let It Be - 1970
            <br />
            <button
              class="btn btn-warning"
              type="button"
            >
              Edit music
            </button>
            <button
              class="btn btn-info"
              type="button"
            >
              View music
            </button>
            <button
              class="btn btn-danger"
              type="button"
            >
              Delete
            </button>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="upperPart"
        style="margin-left: 10px; margin-bottom: 15px;"
      >
        <h1>
          Music forum
        </h1>
        <h5>
          Add, Modify, Remove melodies!
        </h5>
        <div>
          <button
            class="btn btn-outline-primary"
            type="button"
          >
            Add music!
          </button>
           
          <br />
          <button
            class="btn btn-outline-info"
            type="button"
          >
            Check Stats!
          </button>
           
          <br />
          <input
            id="filter"
            name="filter"
            placeholder="Filter by title!"
            type="text"
            value=""
          />
        </div>
      </div>
      <ul
        class="list-group"
        data-testid="list"
      >
        <li
          class="list-group-item list-group-item-dark"
        >
          The Beatles: Rocky Racoon - 1966
          <br />
          <button
            class="btn btn-warning"
            type="button"
          >
            Edit music
          </button>
          <button
            class="btn btn-info"
            type="button"
          >
            View music
          </button>
          <button
            class="btn btn-danger"
            type="button"
          >
            Delete
          </button>
        </li>
        <li
          class="list-group-item list-group-item-dark"
        >
          The Beatles: Nowhere Man - 1965
          <br />
          <button
            class="btn btn-warning"
            type="button"
          >
            Edit music
          </button>
          <button
            class="btn btn-info"
            type="button"
          >
            View music
          </button>
          <button
            class="btn btn-danger"
            type="button"
          >
            Delete
          </button>
        </li>
        <li
          class="list-group-item list-group-item-dark"
        >
          The Beatles: Let It Be - 1970
          <br />
          <button
            class="btn btn-warning"
            type="button"
          >
            Edit music
          </button>
          <button
            class="btn btn-info"
            type="button"
          >
            View music
          </button>
          <button
            class="btn btn-danger"
            type="button"
          >
            Delete
          </button>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
